src/Compilers/CSharp/Portable/Symbols/TypeWithAnnotations.cs

```diff
@@ -1022,4 +1026,83 @@ internal override bool TypeSymbolEquals(TypeWithAnnotations type, TypeWithAnnota
             }
         }
     }
+
+    /// <summary>
+    /// Represents a sequence of <see cref="NullableAnnotation"/> to apply to a type. The sequence of
+    /// flags is specific to how the individual <see cref="TypeSymbol"/> walk their nested types.
+    /// </summary>
+    internal sealed class NullableTransformStream
+    {
+        private static readonly ObjectPool<NullableTransformStream> Pool = new ObjectPool<NullableTransformStream>(() => new NullableTransformStream(), Environment.ProcessorCount);
+
+        private ImmutableArray<byte> _transforms;
+        private int _positionOrDefault;
+
```

Comment:
`/// <summary>If _transforms.IsDefault, contains the transform value to yield an infinite sequence of.  Otherwise is the position in _transforms of the next value to yield.</summary>`